### PR TITLE
Support DataStream APIs

### LINKED ISSUE #402: [FEATURE] Data Stream Api
### Is your feature request related to a problem?
There is data stream functionality of Open Search https://opensearch.org/docs/latest/im-plugin/data-streams/. This functionality cant be used with Open Search Java client.

### What solution would you like?

1. Add data stream api support to Open Search Java client. Something like:
- `client.indices().getDataStream()`
- `client.indices().putDataStream()`
- `client.indices().deleteDataStream()`
2. Add an ability to override `timestamp_field` for `dataStream` at `putIndexTemplate`

### What alternatives have you considered?
Writing my own implementation using `org.opensearch.client.RestClient`

### PR DESCRIPTION
**Pull request by tanqiuliu**
### Description
- Added data stream APIs (`CreateDataStream`, `DeleteDataStream`, `GetDataStream`, `DataStreamsStats`)
- Allowed specifying `timestamp_field` for `dataStream` in `PutIndexTemplate`
- Added integration tests for data stream APIs

### Issues Resolved
#402 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tanqiuliu** [comment]
The test failures are all because of the following test
```
org.opensearch.client.opensearch.integTest.restclient.AggregationRequestIT > testDateRangeAggregation FAILED
    org.opensearch.client.opensearch._types.OpenSearchException at AggregationRequestIT.java:94
```
This is not relevant to the changes in this PR. The same test failure also appeared in previous PRs: [link1](https://github.com/opensearch-project/opensearch-java/actions/runs/4483010005/jobs/7881768054), [link2](https://github.com/opensearch-project/opensearch-java/actions/runs/4461175657/jobs/7836362570). 

When I try to reproduce the issue in local, it seems to be an intermittent issue which appears once among ten runs. It failed due to exception when search an index: 
```
Request failed: [search_phase_execution_exception] all shards failed
```
I am not quite sure about the root cause.

**Comment by reta** [comment]
> I am not quite sure about the root cause.

That's the flaky one sadly, https://github.com/opensearch-project/opensearch-java/issues/407

**Review by reta — DISMISSED** [review]

**Review by VachaShah — APPROVED** [review]
LGTM, thank you @tanqiuliu for adding this feature! 

Can we also add this to the [USER_GUIDE](https://github.com/opensearch-project/opensearch-java/blob/main/USER_GUIDE.md)?

File: USER_GUIDE.md [hunk 1]
```diff
@@ -214,6 +214,53 @@ DeleteIndexRequest deleteIndexRequest = new DeleteRequest.Builder().index(index)
 DeleteIndexResponse deleteIndexResponse = client.indices().delete(deleteIndexRequest);
 ```
 
+## Data Stream API
```
Comment (VachaShah) [inline USER_GUIDE.md:217]: Sorry one more comment: can you add the new section to TOC as well?

Comment (tanqiuliu) [reply USER_GUIDE.md:222]: Added

**Review by reta — APPROVED** [review]

**Review by VachaShah — APPROVED** [review]

**Pull request merged**